MyToDo/App/Storage/CoreDataStorage.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  CoreDataStorage.swift
+//  MyToDo
+//
+//  Created by Ivan Ruiz Monjo on 12/2/21.
+//
+
+import CoreData
+
+final class CoreDataStorage: Storage {
+
+    static let shared = CoreDataStorage()
+    private let persistentContainer: NSPersistentContainer
+    var viewContext: NSManagedObjectContext { persistentContainer.viewContext }
+
+    init() {
+        persistentContainer = NSPersistentContainer(name: String(describing: CoreDataStorage.self))
+        persistentContainer.loadPersistentStores { persistentStoreDescription, error in
+            guard error == nil else {
+                fatalError("Uneable to load. \(error?.localizedDescription ?? "")")
+            }
+        }
+    }
+
+    func save() {
+        if viewContext.hasChanges {
+            do {
+                try viewContext.save()
+            } catch {
+                print("Uneable to save viewContext. \(error.localizedDescription)")
+            }
+        }
+    }
+
+    func delete(object: Any) {
+        guard let coredataObject = object as? NSManagedObject else {
+            return
+        }
+        viewContext.delete(coredataObject)
+        save()
+    }
+
+    // MARK: ToDo
+
+    func getToDos() -> [ToDo] {
```

Comment:
I like adding these in a seperate file that contains an extension, like `CoreDataStorage+ToDo` for simple cases.

In more complex cases you could leverage generics and protocols to do something like this:

```
protocol Fetchable {
  static func fetchRequest() -> NSFetchRequest<Self>
}

extension ToDo: Fetchable {}

extension CoreDataStorage {
  func getItems<T: Fetchable>() -> [T] {
    let request: NSFetchRequest<T> = T.fetchRequest()
    do {
      return try viewContext.fetch(request)
    } catch {
    // ... print / handle error
    }

    return []
  }
}
```

I didn't run this through the compiler but if you'd like we can work through this a bit more on Friday.